src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs

```diff
@@ -361,74 +355,19 @@ public bool ForEach(KeyValuePair<string, object> activityTag)
                     this.Created = true;
                 }
 
-                OtlpCommon.ArrayValue arrayValue;
+                var attribute = activityTag.ToOtlpAttribute();
```

Comment:
the peer serviceResolver part is another switch..case. It might be better to switch to simpler If..Else.

Reply:
Not sure this is what you had in mind, but refactored it a bit to only to the peer service stuff if the attribute was successfully transformed.

Reply:
changed to if/else